src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.cs

```diff
@@ -1,332 +1,123 @@
-                    semanticModel,
-                    cancellationToken);
+                var elseTrailingTrivia = ifNode.Else.GetTrailingTrivia();
+                var ifTrailingTrivia = ifNode.Statement.GetTrailingTrivia();
+                newIfNodeStatement = ifNode.Else.Statement.WithTrailingTrivia(ifTrailingTrivia);
+                newElseStatement = ifNode.Else.WithStatement(ifNode.Statement).WithTrailingTrivia(elseTrailingTrivia);
             }
 
-            return canSimplify;
-        }
-
-        private bool CanSimplifyToLengthEqualsZeroExpression(
-            ExpressionSyntax variableExpression,
-            LiteralExpressionSyntax numericLiteralExpression,
-            SemanticModel semanticModel,
-            CancellationToken cancellationToken)
-        {
-            var numericValue = semanticModel.GetConstantValue(numericLiteralExpression, cancellationToken);
-            if (numericValue.HasValue && numericValue.Value is int && (int)numericValue.Value == 0)
-            {
-                var symbol = semanticModel.GetSymbolInfo(variableExpression, cancellationToken).Symbol;
+            var newIfStatment = ifNode.Else.Statement.Kind() == SyntaxKind.IfStatement && newIfNodeStatement.Kind() != SyntaxKind.Block
+                        ? SyntaxFactory.Block(newIfNodeStatement)
+                        : newIfNodeStatement;
 
-                if (symbol != null && (symbol.Name == "Length" || symbol.Name == "LongLength"))
-                {
-                    var containingType = symbol.ContainingType;
-                    if (containingType != null &&
-                        (containingType.SpecialType == SpecialType.System_Array ||
-                         containingType.SpecialType == SpecialType.System_String))
-                    {
-                        return true;
-                    }
-                }
+            ifNode = ifNode.WithCondition((ExpressionSyntax)(Negate(ifNode.Condition, generator, syntaxFacts, model, cancellationToken)))
+                .WithStatement(newIfStatment)
+                .WithElse(newElseStatement);
 
-                var typeInfo = semanticModel.GetTypeInfo(variableExpression, cancellationToken);
-                if (typeInfo.Type != null)
-                {
-                    switch (typeInfo.Type.SpecialType)
-                    {
-                        case SpecialType.System_Byte:
-                        case SpecialType.System_UInt16:
-                        case SpecialType.System_UInt32:
-                        case SpecialType.System_UInt64:
-                            return true;
-                    }
-                }
+            if (isMultiLine)
+            {
+                ifNode = ifNode.WithAdditionalAnnotations(Formatter.Annotation);
             }
 
-            return false;
+            // get new root
+            return model.SyntaxTree.GetRoot().ReplaceNode(ifStatement, ifNode);
```

Comment:
GetRoot should take the cancellation token.

Reply:
Why not just return the new if statement instead of updating the root?  The caller can update the root themselves.  Now this type is just responsible for figuring out how to do the inversion.